src/viam/sdk/robot/client.cpp

```diff
@@ -253,21 +254,21 @@ std::shared_ptr<RobotClient> RobotClient::with_channel(ViamChannel channel, Opti
 
 std::shared_ptr<RobotClient> RobotClient::at_address(std::string address, Options options) {
     const char* uri = address.c_str();
-    ViamChannel channel = ViamChannel::dial(uri, options.dial_options);
+    auto channel = ViamChannel::dial(uri, options.dial_options);
     std::shared_ptr<RobotClient> robot = RobotClient::with_channel(channel, options);
     robot->should_close_channel = true;
 
     return robot;
 };
 
 std::shared_ptr<RobotClient> RobotClient::at_local_socket(std::string address, Options options) {
-    address = "unix://" + address;
-    const char* uri = address.c_str();
+    std::string addr = "unix://" + address;
+    const char* uri = addr.c_str();
     std::shared_ptr<grpc::Channel> channel =
         grpc::CreateChannel(uri, grpc::InsecureChannelCredentials());
     std::unique_ptr<viam::robot::v1::RobotService::Stub> st =
         viam::robot::v1::RobotService::NewStub(channel);
-    ViamChannel viam_channel = ViamChannel(channel, uri, nullptr);
+    auto viam_channel = std::make_shared<ViamChannel>(channel, address.c_str(), nullptr);
```

Comment:
this was a fun one! we were passing an updated `addr` with the `unix://` prefix to the rust_utils `free_string` function. But! `free_string` calls rust's `c_str::from_raw`, which must take as an argument a string that was previously created by a call to `c_str::into_raw`. So, the `unix://` prefix was breaking things.